src/cata_tiles.cpp

```diff
@@ -1,69 +1,375 @@
+        index_bg = 8,
+    };
+
+    enum : char { default_value = -1 };
+
+    static void set_fg_color(char const n = default_value) { id_[index_fg] = n; }
+    static void set_fg_color(nc_color const col) {
+        id_[index_fg] = static_cast<char>(
+            colorpairs[(col & A_COLOR) >> 17].FG + (col & A_BOLD ? 8 : 0));
+    }
+
+    static void set_bg_color() { id_[index_bg] = default_value; }
+    static void set_bg_color(nc_color const col) {
+        id_[index_bg] = static_cast<char>(
+            colorpairs[(col & A_COLOR) >> 17].BG + (col & A_BLINK ? 8 : 0));
+    }
+
+    static void set_codepoint(long const sym) {
+        id_[index_cp] = static_cast<char>(sym);
+    }
+
+    static std::string const& id() { return id_; }
+private:
+    static std::string id_;
 };
 
-cata_tiles::cata_tiles(SDL_Renderer *render)
+// X is replaced by ascii code (converted to char)
+// F is replaced by foreground (converted to char)
+// B is replaced by background (converted to char)
+std::string ascii_id::id_ {"ASCII_XFB"};
+
+std::string const& to_string(multitile_type const mt)
 {
-    //ctor
-    renderer = render;
+    constexpr int size = static_cast<int>(multitile_type::num_multitile_types);
+    static std::array<std::string, size> const strings {{
+        "center",
+        "corner",
+        "edge",
+        "t_connection",
+        "end_piece",
+        "unconnected",
+        "open",
+        "broken",
+    }};
+
+    auto const i = static_cast<int>(mt);
+    if (i < 0 || i >= size) {
+        dbg(D_ERROR) << "bad multitile_type value";
+    }
+
+    return strings[i];
+}
 
-    tile_height = 0;
-    tile_width = 0;
-    tile_ratiox = 0;
-    tile_ratioy = 0;
+multitile_type from_string(std::string const& s)
+{
+    static std::map<std::string, multitile_type> const map {
+        {"center",       multitile_type::center},
+        {"corner",       multitile_type::corner},
+        {"edge",         multitile_type::edge},
+        {"t_connection", multitile_type::t_connection},
+        {"end_piece",    multitile_type::end_piece},
+        {"unconnected",  multitile_type::unconnected},
+        {"open",         multitile_type::open},
+        {"broken",       multitile_type::broken},
+    };
 
-    in_animation = false;
-    do_draw_explosion = false;
-    do_draw_bullet = false;
-    do_draw_hit = false;
-    do_draw_line = false;
-    do_draw_weather = false;
-    do_draw_sct = false;
-    do_draw_zones = false;
+    auto const it = map.find(s);
+    if (it == map.end()) {
+        dbg(D_ERROR) << "bad multitile_type string";
+        return static_cast<multitile_type>(0);
+    }
 
-    boomered = false;
-    sight_impaired = false;
-    bionight_bionic_active = false;
+    return it->second;
+}
+
+std::unique_ptr<SDL_Surface> create_tile_surface(int const tile_width, int const tile_height)
+{
+    constexpr bool big_e = SDL_BYTEORDER == SDL_BIG_ENDIAN;
+    enum : Uint32 {
+        r_mask = big_e ? 0xFF000000 : 0x000000FF,
```

Comment:
Out of curiosity: is there any difference between this and four separate `constexpr int` values?